x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/Criterion.java

```diff
@@ -89,7 +91,13 @@ public Ordinal ordinal(SearchHit hit) {
             }
             tbreaker = (Comparable<Object>) tb;
         }
-        return new Ordinal(timestamp, tbreaker);
+
+        Object implicitTbreaker = implicitTiebreaker.extract(hit);
+        if (implicitTbreaker instanceof Number == false) {
+            throw new EqlIllegalArgumentException("Expected _shard_doc/implicit tiebreaker as long but got [{}]", implicitTbreaker);
```

Comment:
Regarding the error message, could the value be a non-long integer?

Reply:
@bpintea no. Always `long`.

Reply:
Would then the instance check agains a `Long` be also appropriate? (just FMI, its prolly slightly safer the way it is now)

Reply:
I'd leave it as is, for the reason you mentioned but also to be consistent with the approach throughout that method.